packages/flutter/lib/src/rendering/object.dart

```diff
@@ -194,9 +248,10 @@ class PaintingContext extends ClipContext {
     assert(_canvas == null || _canvas!.getSaveCount() == 1);
 
     // Create a layer for our child, and paint the child into it.
-    if (child._needsPaint) {
+    if (child._needsPaint || !child._wasRepaintBoundary) {
       repaintCompositedChild(child, debugAlsoPaintedParent: true);
     } else {
+      updateLayerProperties(child);
```

Comment:
While there is no harm in calling this unconditionally, should we actually only call it if the child is dirty for _needsCompositedLayerUpdate?

Reply:
Ahh, but this is a RenderObject, not a RenderBox, so the child is not accessible.

Reply:
oh I see what you mean - hhmm